src/components/PostPageDetail/index.js

```diff
@@ -577,6 +581,11 @@ const PostPageDetailIdComponent = (props) => {
             isBackButton={true}
             source={SOURCE_PDP}
             height={getHeightHeader()}
+            isFollow={item?.is_following_target || parentData?.is_following_target}
+            onPressFollUnFoll={() => {
+              followUnfollow(item);
+              setItem({...item, is_following_target: !item?.is_following_target});
+            }}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This code block introduces a potential race condition. The `followUnfollow` function is asynchronous and may take some time to complete, but the state update for `setItem` happens immediately after calling `followUnfollow`. If `followUnfollow` fails or takes longer than expected, the UI will show incorrect follow/unfollow status. Consider awaiting the `followUnfollow` function before updating the state.
```diff
- onPressFollUnFoll={() => {
-   followUnfollow(item);
-   setItem({...item, is_following_target: !item?.is_following_target});
- }}
+ onPressFollUnFoll={async () => {
+   const result = await followUnfollow(item);
+   if(result) {
+     setItem({...item, is_following_target: !item?.is_following_target});
+   }
+ }}
```
Please note that you should handle errors appropriately in the `followUnfollow` function to ensure `result` is `true` only when the operation is successful.


<!-- This is an auto-generated comment by OSS CodeRabbit -->